compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -0,0 +1,196 @@
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex mapVertex = new OperatorVertex(new MapTransform<>(func));
+    mapVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(mapVertex, loopVertexStack);
+
+    final IREdge newEdge = new IREdge(getEdgeCommunicationPattern(lastVertex, mapVertex),
+        lastVertex, mapVertex, new SparkCoder(kryoSerializer));
+    newEdge.setProperty(KeyExtractorProperty.of(new SparkKeyExtractor()));
+    builder.connectVertices(newEdge);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism, builder.buildWithoutSourceSinkCheck(), mapVertex);
+  }
+
+  /////////////// ACTIONS ///////////////
+
+  private static final AtomicInteger RESULT_ID = new AtomicInteger(0);
+
+  /**
+   * This method is to be removed after a result handler is implemented.
+   * @return a unique integer.
+   */
+  public static Integer getResultId() {
+    return RESULT_ID.getAndIncrement();
+  }
+
+  /**
+   * Reduce action.
+   * @param func function (binary operator) to apply.
+   * @return the result of the reduce action.
+   */
+  public T reduce(final SerializableBinaryOperator<T> func) {
+    // save result in a temporary file
+    final String resultFile = System.getProperty("user.dir") + "/reduceresult";
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex reduceVertex = new OperatorVertex(new ReduceTransform<>(func, resultFile));
+    reduceVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(reduceVertex, loopVertexStack);
+
+    final IREdge newEdge = new IREdge(getEdgeCommunicationPattern(lastVertex, reduceVertex),
+        lastVertex, reduceVertex, new SparkCoder(kryoSerializer));
+    newEdge.setProperty(KeyExtractorProperty.of(new SparkKeyExtractor()));
+    builder.connectVertices(newEdge);
+
+    // launch DAG
+    JobLauncher.launchDAG(builder.build());
+
+    // Retrieve result data from file.
+    try {
+      final Kryo kryo = new Kryo();
+      final List<T> result = new ArrayList<>();
+      Integer i = 0;
+      File file = new File(resultFile + i);
+      while (file.exists()) {
+        final Input input = new Input(new FileInputStream(resultFile + i));
+        result.add((T) kryo.readClassAndObject(input));
+        input.close();
+
+        // Delete temporary file
+        file.delete();
+        file = new File(resultFile + ++i);
+      }
+      return result.stream().reduce(func).get();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /////////////// MISC ///////////////
+
+  /**
+   * Retrieve communication pattern of the edge.
+   * @param src source vertex.
+   * @param dst destination vertex.
+   * @return the communication pattern.
+   */
+  private static DataCommunicationPatternProperty.Value getEdgeCommunicationPattern(final IRVertex src,
```

Comment:
This method is misleading in that it takes as arguments two vertices as src/dst, but always return `OneToOne`. I'd remove the method.

Reply:
This has been added to further use other data communication patterns in the upcoming PRs, when supporting a more variety of transformations and actions. Ill instead add a todo note in the method.